### PR TITLE
mgmt, migrate azure-resourcemanager-resources to azure-json

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

- part of https://github.com/Azure/azure-sdk-for-java/issues/41194
- manual code: https://github.com/Azure/azure-sdk-for-java/commit/b4f3d57ff3a4ac90f3d2f81c35b978630bfb9515
- minor fix for final class: https://github.com/Azure/azure-sdk-for-java/pull/41277/commits/7fb540b69d6ffc01412f98ab460cb37a6ffa9bbb
  same with https://github.com/Azure/azure-sdk-for-java/pull/40410

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/api-specs.json [hunk 1]
```diff
@@ -3,86 +3,86 @@
...
+    "args": "--tag=profile-hybrid-2020-09-01 --preserve-model=ContainerServiceVMSizeTypes --stream-style-serialization=false"
   },
   "cosmos": {
     "dir": "azure-resourcemanager-cosmos",
     "source": "specification/cosmos-db/resource-manager/readme.md",
     "package": "com.azure.resourcemanager.cosmos",
-    "args": "--tag=package-2024-05"
+    "args": "--tag=package-2024-05 --stream-style-serialization=false"
   },
   "deploymentstacks": {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/api-specs.json:87]: deploymentstacks feature etc. also belong to resources lib, no change here?

see https://github.com/weidongxu-microsoft/work-notes/blob/main/fluent-v2-generator-usage.md#script-to-generate-code-for-fluent-premium

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/DeploymentExportResultImpl.java [hunk 2]
```diff
@@ -17,11 +20,11 @@ final class DeploymentExportResultImpl extends
         implements
         DeploymentExportResult {
 
-    private ObjectMapper mapper;
+    private SerializerAdapter serializerAdapter;
 
     DeploymentExportResultImpl(DeploymentExportResultInner innerModel) {
         super(innerModel);
-        mapper = new ObjectMapper();
+        serializerAdapter = SerializerFactory.createDefaultManagementSerializerAdapter();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/DeploymentExportResultImpl.java:27]: Note that a simple ObjectMapper is not exactly same as SerializerAdapter created from azure-core. Latter has lots of configure which is for Azure JSON (e.g. date-time, duration, etc.).

ObjectMapper more like the innerMapper in https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java#L103

Maybe not a big deal here, as the ARM template should be simple JSON with string/numeric etc. It should not involve any advanced type like DateTime.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/DeploymentExportResultImpl.java:27]: We may need advice from Alan on whether `SerializerAdapter` is still the recommended class for serialization/de-serialization.

File: sdk/resourcemanager/api-specs.json (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/api-specs.json:87]: No manual code change there.
I used `gulp codegen --projects locks,features,changes,deploymentstacks,resources,subscriptions,policy --preserve` to regen. Is it OK?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/api-specs.json:87]: I see, you've remove the option from gulpfile.js

Then that is fine, the default is true.

Comment (XiaofeiCao) [reply sdk/resourcemanager/api-specs.json:87]: updated

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/DeploymentExportResultImpl.java (hunk 2 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/implementation/DeploymentExportResultImpl.java:27]: Yeah. I'm seeing Alan's PR doing that:
https://github.com/Azure/azure-sdk-for-java/pull/40716/files#diff-3f1e47c051070c3f2d4a385d788c972fa02611ca0f9c9cf2652e40eefa45a683R30-R31

And the serializerAdapter is `JacksonAdapter`:
https://github.com/Azure/azure-sdk-for-java/pull/40716/files#diff-1d04b9c2cc5bc129be708cce645415ee0a8f8489515563b8ab21e0be132a96b0R102

@alzimmermsft Would you help confirm that we can substitute `ObjectMapper` for `JacksonAdapter`? And is this PR using it the correct way?

**Review by alzimmermsft — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**